plugins/primus_lisp/semantics/bits.lisp

```diff
@@ -19,13 +19,87 @@
          (logand (msb rm) (lnot (msb rd)))
          (logand (msb rn) (lnot (msb rd)))))
 
-
 (defun overflow (rd rn rm)
   "(overflow RD RN RM) is true if the sum RD = RN + RM results
    in two's complement overflow."
   (logor (logand (msb rn) (msb rm) (lnot (msb rd)))
          (logand (lnot (msb rn)) (lnot (msb rm)) (msb rd))))
 
+(defun highest-set-bit (bitv)
```

Comment:
We now have `clzN` in `init.lisp` that is implemented using branchless/loopless algorithm (but only works on immediates of up to 64 bits. You can use it, instead. If you think that we can have larger immediates, then we need to implement a generic clz that depending on the word-size either call the optimized versions or fall-back to the while-loop version.

Reply:
ok, the immediates should only be at most 64 bits wide.